Error: Only named exports may use 'export type'.

Hi,

I just upgraded to the newest version `0.3.21` and got the following error when trying to build my project:

```
> tsc -p tsconfig.package.json

../../node_modules/@jridgewell/trace-mapping/dist/types/trace-mapping.d.ts:4:1 - error TS1383: Only named exports may use 'export type'.

4 export type * from './types';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in ../../node_modules/@jridgewell/trace-mapping/dist/types/trace-mapping.d.ts:4
```

I think it´s related to this commit https://github.com/jridgewell/trace-mapping/commit/d04e7fe0bda0952e1542dd89a6ae3ff85967346a

After downgrading back to `0.3.20` the error is gone. Any ideas?

--- a/src/trace-mapping.ts
+++ b/src/trace-mapping.ts
@@ -41,7 +41,25 @@ import type { Source } from './by-source';
 import type { MemoState } from './binary-search';
 
 export type { SourceMapSegment } from './sourcemap-segment';
-export type * from './types';
+export type {
+  SourceMap,
+  DecodedSourceMap,
+  EncodedSourceMap,
+  Section,
+  SectionedSourceMap,
+  SourceMapV3,
+  Bias,
+  EachMapping,
+  GeneratedMapping,
+  InvalidGeneratedMapping,
+  InvalidOriginalMapping,
+  Needle,
+  OriginalMapping,
+  OriginalMapping as Mapping,
+  SectionedSourceMapInput,
+  SourceMapInput,
+  SourceNeedle,
+} from './types';
 
 const LINE_GTR_ZERO = '`line` must be greater than 0 (lines start at line 1)';
 const COL_GTR_EQ_ZERO = '`column` must be greater than or equal to 0 (columns start at column 0)';

